src/model/SetpointManagerSystemNodeResetHumidity.cpp

```diff
@@ -0,0 +1,257 @@
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "SetpointManagerSystemNodeResetHumidity.hpp"
+#include "SetpointManagerSystemNodeResetHumidity_Impl.hpp"
+
+#include "Node.hpp"
+#include "Node_Impl.hpp"
+#include "PlantLoop.hpp"
+#include "Model.hpp"
+
+#include <utilities/idd/IddFactory.hxx>
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/OS_SetpointManager_SystemNodeReset_Humidity_FieldEnums.hxx>
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+  namespace detail {
+
+    SetpointManagerSystemNodeResetHumidity_Impl::SetpointManagerSystemNodeResetHumidity_Impl(const IdfObject& idfObject, Model_Impl* model,
+                                                                                             bool keepHandle)
+      : SetpointManager_Impl(idfObject, model, keepHandle) {
+      OS_ASSERT(idfObject.iddObject().type() == SetpointManagerSystemNodeResetHumidity::iddObjectType());
+    }
+
+    SetpointManagerSystemNodeResetHumidity_Impl::SetpointManagerSystemNodeResetHumidity_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                                                                             Model_Impl* model, bool keepHandle)
+      : SetpointManager_Impl(other, model, keepHandle) {
+      OS_ASSERT(other.iddObject().type() == SetpointManagerSystemNodeResetHumidity::iddObjectType());
+    }
+
+    SetpointManagerSystemNodeResetHumidity_Impl::SetpointManagerSystemNodeResetHumidity_Impl(const SetpointManagerSystemNodeResetHumidity_Impl& other,
+                                                                                             Model_Impl* model, bool keepHandle)
+      : SetpointManager_Impl(other, model, keepHandle) {}
+
+    const std::vector<std::string>& SetpointManagerSystemNodeResetHumidity_Impl::outputVariableNames() const {
```

Comment:
Need to override this if it's empty anyway?

Reply:
You do. This is a pure virtual method in ModelObject, so you must define it.

https://github.com/NREL/OpenStudio/blob/2924b53962e70d425a77f677dfcf74de1f1458af/src/model/ModelObject_Impl.hpp#L132